src/stem.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+var util = require("./util");
+
+
+/**
+ * Stem
+ *
+ * A class that creates and manages the "insert widget here" buttons a.k.a "stems"
+ * that appear beside an empty block level element.
+ */
+
+function Stem(blockElem, onClick) {
+  util.assert(!blockElem.$stem);
+
+  this.blockElem = blockElem;
+  this.blockElem.$stem = this;
+
+  this.stemButton = this.createDom();
+  this.reposition();
+
+  // Handle click event
+  this.stemButton.addEventListener( 'click', function() {
+    onClick();
+  }, false);
+
+}
+
+// Static "idompotent" function to retrive or create a Stem
+Stem.getOrCreate = function(blockElem, onClick) {
+  return blockElem.$stem || new Stem(blockElem, onClick);
+}
+
+
+// Hide the stem
+Stem.prototype.destroy = function() {
+  // Delete back reference
+  delete this.blockElem.$stem;
+  // Remove DOM
+  document.body.removeChild( this.stemButton );
```

Comment:
use 

util.removeNode(the elem)

reasons
- less boilerplate
- doesn't depend on the fact it's attached to document.body, which we'll probably change?